common-crafting.lic

```diff
@@ -241,4 +242,32 @@ module DRCC
       end
     end
   end
+
+  def clean_brazier?
+    case result = DRC.bput('look on brazier', 'There is nothing on there', "On the .* you see")
+    when /There is nothing on there/i
+      true
+    when /On the .* you see/
+      case DRC.bput("clean brazier", 'You prepare to clean off the brazier', 'There is nothing', 'The brazier is not currently lit')
+      when 'You prepare to clean off the brazier'
+        DRC.bput("clean brazier", 'a massive ball of flame jets forward and singes everything nearby')
+      end
+      empty_brazier
+      true
+    end
+  end
+
+  def empty_brazier
+    case DRC.bput("look on brazier", /On the (.*)brazier you see (.*)./, 'There is nothing')
+    when /On the (.*)brazier you see (.*)./
+      items = Regexp.last_match(2)
+      items = items.split(' and ')
+      items.each do |item|
+        item = item.split.last
+        DRC.bput("get #{item} from brazier",'You get')
+        DRCT.dispose(item)
+      end
+    end
+  end
+    
```

Comment:
Since you are adding to common-crafting, are you removing the same defs from `enchant.lic`?

Reply:
Still looking at this, as I wasn't running into that code block in my testing earlier (with society braziers), which if it's in enchant.lic I would have expected.

Reply:
(I think) I found the issue with enchant.lic and cleaning brazier
> [enchant]>put my totem on enchanter's brazier
> You glance down at the smoking brazier and reconsider if you wish to begin enchanting.
> [Putting an item on the brazier begins the enchanting process.  This process can enhance the item, or cause serious damage to it and the enchanter.  Put your item onto the brazier again in the next 30 seconds, to proceed.  You've been warned!]
> > 
> [enchant]>put my totem on enchanter's brazier
> You must first clean off the brazier.
> > 
> [enchant]>clean enchanter's brazier
> The brazier is not currently lit, and so you feel it is safe to collect anything atop it.
> > 
> [enchant]>look on enchanter's brazier
> On the enchanter's brazier you see a bone totem.
> > 
> [enchant]>get totem from brazier
> You get a plain bone totem from atop an enchanter's brazier.
> > 
> [enchant]>put my totem in my duffel bag
> You put your totem in your duffel bag.
> > 
> [enchant]>get my totem from my duffel bag
> You get a plain bone totem from inside your duffel bag.
> > 
> [enchant]>put my totem on enchanter's brazier
> You glance down at the smoking brazier and reconsider if you wish to begin enchanting.
> [Putting an item on the brazier begins the enchanting process.  This process can enhance the item, or cause serious damage to it and the enchanter.  Put your item onto the brazier again in the next 30 seconds, to proceed.  You've been warned!]
> > 
> [enchant]>put my totem on enchanter's brazier
> With a flick, you pop open the brazier's lid.  Tiny motes briefly encircle the almond-shaped flame now burning within the brazier.  Carefully you place your totem atop the brazier.
> Once finished you sense an imbue spell will be required to continue enchanting.
> Roundtime: 14 sec.
> You need another mana fount to continue crafting a plain bone totem.  WAVE a functional mana fount at the a plain bone totem to align the item with the fount's focusing properties.  Then continue the process with the casting of an imbue spell.
> R> 
> Learned: Enchanting(+1:24/34)
> ,pa
> --- Lich: enchant paused.
> R> glance
> **You glance down to see a plain bone totem in your right hand and nothing in your left hand.**

You can end up with an item still in your hand when starting, which causes problems at the end of enchant process.  I think the best way to handle it is the same way handling anvils / etc are handled.

So yeah, I'll work on pulling out the clean code from enchant and handle it in common-crafting only.